common/flatpak-dir.c

```diff
@@ -6527,18 +6533,31 @@ flatpak_dir_check_for_update (FlatpakDir          *self,
 #ifdef FLATPAK_ENABLE_P2P
       /* Find the latest rev from the remote and its available mirrors, including
        * LAN and USB sources. */
+      GVariantBuilder find_builder;
```

Comment:
`g_auto(GVariantBuilder) find_builder = FLATPAK_VARIANT_BUILDER_INITIALIZER`?

Reply:
Just to make sure I understand, the only situation where that would make a difference is if execution didn't make it to the `g_variant_builder_end` call that frees the `GVariantBuilder` memory?

Reply:
Yes, either through the current state of the code, or due to subsequent refactoring.